### PR TITLE
simple client filtering on list_jobs

### PR DESCRIPTION
**Pull request by adelebai**

### CODE REVIEW

**Review by sanjgupt — APPROVED** [review]

**Comment by vxfield** [comment]
I added @guenp as a reviewer since she is back as co-owner of the Azure Quantum SDK for Python.
Please wait for her review before merging this.

File: azure-quantum/azure/quantum/job.py
Comment (guenp) [inline azure-quantum/azure/quantum/job.py:113]:
```suggestion
        if name_match is not None and re.search(name_match, self.details.name) is None:
           return False
```

File: azure-quantum/azure/quantum/job.py
Comment (guenp) [inline azure-quantum/azure/quantum/job.py:117]:
```suggestion
        if status is not None and self.details.status != status.value:
            return False
```

File: azure-quantum/azure/quantum/job.py
Comment (guenp) [inline azure-quantum/azure/quantum/job.py:121]:
```suggestion
        if created_after is not None and self.details.creation_time < created_after:
            return False
```

File: azure-quantum/azure/quantum/workspace.py
Comment (guenp) [inline azure-quantum/azure/quantum/workspace.py:239]: I would recommend not putting todo's in the code but instead track them in ADO and link this file there

**Review by guenp — CHANGES_REQUESTED** [review]

**Review by guenp — APPROVED** [review]

**Pull request merged**
